typings/index.d.ts

```diff
@@ -72,4 +99,18 @@ export type RichTextStyles = {
     | StyleProp<ImageStyle>
 }
 
+export interface SerializerFunction {
+  (
+    type: RichTextElementType,
+    element: SpanType,
+    text: string,
+    children: React.ComponentElement<TextProps, Text>,
+    index: string
+  ): React.ComponentElement<TextProps, Text> | null
```

Comment:
if you type the return type as a Text, you cannot return a <Youtube /> component I guess.
@bitboxer what do you think about this?

Reply:
Ah, yeah...this needs a broader skope then.

Reply:
Short heads up: Will continue working on this Wednesday, Tuesday is a holiday here in Berlin and I took a vacation day on Monday.

Reply:
@Spoutnik97 I am now using `JSX.Element`. Hope this is okay. It basically is a `React.Component` with an any type. Now everyone can return whatever they prefer.

Reply:
Perfect!